src/DAO/UploadDAO.php

```diff
@@ -0,0 +1,220 @@
+
+    /**
+     * @param string $token
+     */
+    public function setToken(string $token){
+        $this->token = $token;
+    }
+
+    /**
+     * @return string
+     */
+    public function getToken() : string
+    {
+        return $this->token;
+    }
+
+    /**
+     * @param int $receivedBytes
+     */
+    public function setReceivedBytes(int $receivedBytes){
+        $this->receivedBytes = $receivedBytes;
+    }
+
+    /**
+     * @return int
+     */
+    public function getReceivedBytes() : int
+    {
+        return $this->receivedBytes;
+    }
+
+    /**
+     * @param int $maxUploadSize
+     */
+    public function setMaxUploadSize(int $maxUploadSize)
+    {
+        $this->maxUploadSize = $maxUploadSize;
+    }
+
+    /**
+     * @return int
+     */
+    public function getMaxUploadSize() : int
+    {
+        return $this->maxUploadSize;
+    }
+
+    /**
+     * @param bool $done
+     */
+    public function setDone(bool $done){
+        $this->done = $done;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isDone() : bool
+    {
+        return $this->done;
+    }
+
+    /**
+     * @param string $identifier
+     */
+    public function setIdentifier(string $identifier){
+        $this->identifier = $identifier;
+    }
+
+    /**
+     * @return string
+     */
+    public function getIdentifier() : string {
+        return $this->identifier;
+    }
+
+    /**
+     * @param string $created
+     */
+    public function setCreated(string $created){
+        $this->created = $created;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCreated(){
+        return $this->created;
+    }
+
+    public function loadDBObject(): ?OODBBean
+    {
+        $uploadBean = R::findOne('upload', 'token = ?', [$this->getToken()]);
+
+        if ($uploadBean instanceof OODBBean) {
+            return $uploadBean;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param $token
+     * @return UploadDAO|null
+     */
+    public static function getUploadFromToken($token): ?UploadDAO
+    {
+        $upload = R::findOne('upload', 'token = ?', [$token]);
+
+        if($upload instanceof OODBBean){
+            return new self(
+                $upload->token,
+                $upload->receivedBytes,
+                $upload->maxUploadSize,
+                $upload->done,
+                $upload->identifier,
+                $upload->created
+            );
+        }
+
+        return null;
+    }
+
+    /**
+     * @param $identifier
+     * @return UploadDAO|null
+     */
+    public static function getUploadFromIdentifier($identifier): ?UploadDAO
+    {
+        $upload = R::findOne('upload', 'identifier = ?', [$identifier]);
+
+        if($upload instanceof OODBBean){
+            return new self(
+                $upload->token,
+                $upload->receivedBytes,
+                $upload->maxUploadSize,
+                $upload->done,
+                $upload->identifier,
```

Comment:
This can be the hash of the guest link or the e-mail of the user who has initiated the upload. It is used to assign an upload to a user or guest link.